src/app/src/test/kotlin/org/vaccineimpact/api/tests/logic/UserLogicTests.kt

```diff
@@ -0,0 +1,35 @@
+package org.vaccineimpact.api.tests.logic
+
+import com.nhaarman.mockito_kotlin.*
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.vaccineimpact.api.app.logic.RepositoriesUserLogic
+import org.vaccineimpact.api.app.repositories.UserRepository
+import org.vaccineimpact.api.security.WebTokenHelper
+import org.vaccineimpact.api.test_helpers.MontaguTests
+import org.vaccineimpact.api.test_helpers.exampleInternalUser
+
+class UserLogicTests : MontaguTests()
+{
+    @Test
+    fun `getUserByUsername passes through to repository`()
+    {
+        val repo = mock<UserRepository>()
+        val sut = RepositoriesUserLogic(repo, mock())
+        sut.getUserByUsername("user")
+        verify(repo).getUserByUsername("user")
+    }
+
+    @Test
+    fun `logInAndGetToken gets token and updates last logged in`()
```

Comment:
Looks like we had a test before that checked that `lastLoggedIn` wasn't updated on a failed login, but that's no longer under test? Unless I've missed it

Reply:
This is still tested, just less directly: In the controller test, we check that the logic method `logInAndGetToken` is not called. And in the logic test we check that calling `logInAndGetToken` sets `lastLoggedIn`.

Reply:
Ah yes, ok